x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/AuthorizationService.java

```diff
@@ -502,10 +503,12 @@ private void authorizeAction(
                                 indicesAndAliasesResolver.resolve(action, request, projectMetadata, authorizedIndices)
                             ),
                             e -> {
-                                auditTrail.accessDenied(requestId, authentication, action, request, authzInfo);
-                                if (e instanceof IndexNotFoundException) {
+                                if (e instanceof IndexNotFoundException
+                                    || e instanceof InvalidIndexNameException
+                                    || e instanceof IllegalArgumentException) {
                                     listener.onFailure(e);
                                 } else {
+                                    auditTrail.accessDenied(requestId, authentication, action, request, authzInfo);
```

Comment:
In my opinion, it makes no sense to audit `400` and `404` responses. Previously, we would audit `access_denied` and return `404` response.